### PR TITLE
docs: clarify documentation for matchPaths

### PR DESCRIPTION
**Pull request by fgreinacher**
## Changes

Clarifies the documentation around `matchPaths` and adds a more complex example.

## Context

See https://github.com/renovatebot/renovate/issues/17642#issuecomment-1335053583

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/usage/configuration-options.md [hunk 1]
```diff
@@ -1893,17 +1893,18 @@ Just like the earlier `matchPackagePatterns` example, the above will configure `
 
 ### matchPaths
 
-Renovate will match `matchPaths` against both a partial string match or a minimatch glob pattern.
-If you want to avoid the partial string matching so that only glob matching is performed, wrap your string in `+(...)` like so:
+Renovate will match `matchPaths` against a minimatch glob pattern.
+
+For example the following would match any `package.json`:
```
Comment (rarkins) [inline docs/usage/configuration-options.md:1898]:
```suggestion
For example the following would match any `package.json`, including files like `backend/package.json`:
```

File: docs/usage/configuration-options.md [hunk 2]
```diff
@@ -1893,17 +1893,18 @@ Just like the earlier `matchPackagePatterns` example, the above will configure `
...
 
 ```
-  "matchPaths": ["+(package.json)"],
+  "matchPaths": ["package.json"],
 ```
 
-The above will match only the root `package.json`, whereas the following would match any `package.json` in any subdirectory too:
+While the following would match any file in directories starting with `app-`:
```
Comment (rarkins) [inline docs/usage/configuration-options.md:1904]:
```suggestion
The following would match any file in directories starting with `app-`:
```

**Review by rarkins — CHANGES_REQUESTED** [review]

File: docs/usage/configuration-options.md [hunk 3]
```diff
@@ -1893,17 +1893,18 @@ Just like the earlier `matchPackagePatterns` example, the above will configure `
 
 ### matchPaths
 
-Renovate will match `matchPaths` against both a partial string match or a minimatch glob pattern.
-If you want to avoid the partial string matching so that only glob matching is performed, wrap your string in `+(...)` like so:
+Renovate will match `matchPaths` against a minimatch glob pattern.
```
Comment (HonkingGoose) [inline docs/usage/configuration-options.md:1896]: > Renovate will match `matchPaths` against a minimatch glob pattern.

This sentence has the word `match` in it three times:

- `match`
- `matchPaths`
- `minimatch`

How about this?

```suggestion
Renovate finds the file(s) listed in `matchPaths` with a minimatch glob pattern.
```

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

File: docs/usage/configuration-options.md [hunk 4]
```diff
@@ -1893,17 +1893,18 @@ Just like the earlier `matchPackagePatterns` example, the above will configure `
...
-Renovate will match `matchPaths` against both a partial string match or a minimatch glob pattern.
-If you want to avoid the partial string matching so that only glob matching is performed, wrap your string in `+(...)` like so:
+Renovate finds the file(s) listed in `matchPaths` with a minimatch glob pattern.
 
-```
-  "matchPaths": ["+(package.json)"],
+For example the following would match any `package.json`, including files like `backend/package.json`:
+
+```json
+  "matchPaths": ["package.json"],
```
Comment (HonkingGoose) [inline docs/usage/configuration-options.md:1901]:
```suggestion
{
  "packageRules": [
    {
      "description": "Find package.json files and do INSERT EXPLANATION",
      "matchPaths": ["package.json"],
    }
  ]
}
```

File: docs/usage/configuration-options.md [hunk 5]
```diff
@@ -1893,17 +1893,18 @@ Just like the earlier `matchPackagePatterns` example, the above will configure `
...
 ```
 
-The above will match only the root `package.json`, whereas the following would match any `package.json` in any subdirectory too:
+The following would match any file in directories starting with `app-`:
 
-```
-  "matchPaths": ["package.json"],
+```json
+  "matchPaths": ["app-*/**"],
```
Comment (HonkingGoose) [inline docs/usage/configuration-options.md:1907]:
```suggestion
{
  "packageRules": [
    {
      "description": "Match any file in directories starting with app- and do INSERT EXPLANATION",
      "matchPaths": ["app-*/**"],
    }
  ]
}
```

**Review by HonkingGoose — CHANGES_REQUESTED** [review]
We should use valid JSON. 😉 

The docs say `matchPaths` must have `packageRules` as a parent.

What's a common usecase for setting `matchPaths`? We should insert that into the code example so it makes sense.

File: docs/usage/configuration-options.md (hunk 4 above)
Comment (fgreinacher) [reply docs/usage/configuration-options.md:1901]: I'd say this is mostly used to group dependencies somehow, does this help?

```suggestion
"matchPaths": ["package.json"],
{
  "packageRules": [
    {
      "description": "Group dependencies from package.json files",
      "matchPaths": ["**/package.json"],
      "groupName": "All package.json changes "
    }
  ]
}
```

Comment (fgreinacher) [reply docs/usage/configuration-options.md:1901]: (I also changed `package.json` to `**/package.json` to make clear that this needs to be a pattern)

File: docs/usage/configuration-options.md (hunk 5 above)
Comment (fgreinacher) [reply docs/usage/configuration-options.md:1907]: This is very useful in monorepos where you want to have separate PRs per component. Is this better?

```suggestion
{
  "packageRules": [
    {
      "description": "Group all dependencies from the app directory",
      "matchPaths": ["app/**"],
      "groupName": "App dependencies" 
    }
  ]
}
```

File: docs/usage/configuration-options.md (hunk 4 above)
Comment (HonkingGoose) [reply docs/usage/configuration-options.md:1901]:
```json

File: docs/usage/configuration-options.md [hunk 6]
```diff
@@ -1893,17 +1893,34 @@ Just like the earlier `matchPackagePatterns` example, the above will configure `
...
+      "description": "Group dependencies from package.json files",
+      "matchPaths": ["**/package.json"],
+      "groupName": "All package.json changes"
+    }
+  ]
+}
 ```
 
-The above will match only the root `package.json`, whereas the following would match any `package.json` in any subdirectory too:
+The following would match any file in directories starting with `app-`:
```
Comment (rarkins) [inline docs/usage/configuration-options.md:1912]:
```suggestion
The following would match any file in directories starting with `app/`:
```

**Review by HonkingGoose — APPROVED** [review]

**Review by rarkins — APPROVED** [review]

**Pull request merged**